### PR TITLE
feat(nats): propagate nats id for JSON-based msgs

### PR DESCRIPTION
**Pull request by idonava**
1. add epsagon to nats messages only if: 
   a. process.env.EPSAGON_PROPAGATE_NATS_ID=TRUE
   b. is a JSON based message.
2. update Readme.

### CODE REVIEW

**Review by ranrib — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -140,6 +140,9 @@ An environment variable must be added to start tracing the following libraries:
 
 dns: `EPSAGON_DNS_INSTRUMENTATION = true`
 
+NATS: `EPSAGON_PROPAGATE_NATS_ID = TRUE` Propagate Nats Identifier for JSON-based messages.
```
Comment (ronyis) [inline README.md:143]: Choose if `TRUE` or `true` for both :)

**Review by ronyis — DISMISSED** [review]
One comment

File: README.md (hunk 1 above)
Comment (idonava) [reply README.md:143]: fixed :)

File: src/events/nats.js [hunk 2]
```diff
@@ -55,6 +56,9 @@ const getPublishParams = (subject, msg, opt_reply, opt_callback, jsonConnectProp
         opt_callback_internal = opt_reply;
         opt_reply_internal = undefined;
     }
+    if (jsonConnectProperty && (process.env.EPSAGON_PROPAGATE_NATS_ID || '').toUpperCase() === 'TRUE') {
```
Comment (ronyis) [inline src/events/nats.js:59]: I think we would like to use `TRUE` in default. Please check with @ranrib / @henprtz

**Review by ronyis — APPROVED** [review]

File: src/events/nats.js (hunk 2 above)
Comment (idonava) [reply src/events/nats.js:59]: I talked about it with ran, He told me to use default `FALSE`

File: README.md (hunk 1 above)
Comment (galbash) [reply README.md]: maybe disable only if set to false (reverse behaviour)

Comment (galbash) [reply README.md:143]: maybe disable only if set to false (reverse behavior)

File: src/events/nats.js [hunk 3]
```diff
@@ -55,6 +56,9 @@ const getPublishParams = (subject, msg, opt_reply, opt_callback, jsonConnectProp
...
+        msg_internal.epsagon_id = uuid4();
```
Comment (galbash) [inline src/events/nats.js:60]: every message in nats already has an sid, no need to inject a new one (I can point you to the docs)

**Review by galbash — CHANGES_REQUESTED** [review]

**Review by galbash — APPROVED** [review]

**Pull request merged**